packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CodeModelPlugin.cs

```diff
@@ -20,7 +20,7 @@ namespace Microsoft.Generator.CSharp
     public abstract class CodeModelPlugin
     {
         private static CodeModelPlugin? _instance;
-        internal static CodeModelPlugin Instance
+        public static CodeModelPlugin Instance
```

Comment:
so this is settable, is it safe?
How about we make it like this:
```csharp
public abstract class CodeModelPlugin
{
	public static CodeModelPlugin Instance
	{
		get { return _instance ?? throw new Exception(); }
		private set { _instance = value; }
	}

    [ImportingConstructor]
    public CodeModelPlugin(GeneratorContext context)
    {
		_instance = this;
        Configuration = context.Configuration;
        _inputLibrary = new(() => new InputLibrary(Instance.Configuration.OutputDirectory));
    }
}
```
i think this might work? because the ctor will be invoked by the extension framework therefore the `_instance` will always be set.